chain/client/src/view_client.rs

```diff
@@ -507,7 +504,7 @@ impl ViewClientActor {
             target_shard_id,
             true,
         ) {
-            match self.chain.get_final_transaction_result(&tx_hash) {
+            match self.chain.get_transaction_result_for_rpc(&tx_hash) {
```

Comment:
Consider renaming `get_tx_status` to `get_tx_status_for_rpc` to emphasize what is the purpose of the whole flow.
Or give names `get_partial_tx_status` and `get_partial_transaction_result` to make all methods consistent.